version: '3.9'

services:
  main_db:
    image: postgres:15
    container_name: keykeeper_db
    ports:
      - "5432:5432"
    environment:
      - TZ=GMT+3
      - PGTZ=GMT+3
    env_file:
      - ./postgres/.env
    volumes:
      - ./postgres/.pg/pg_data/:/var/lib/postgresql/data
      - ./postgres/.pg/source/:/var/lib/postgresql/source
    networks:
      - keykeeper-net
  app:
    build: 
      dockerfile: ./flask_api/Dockerfile
      context: .
    container_name: flask_api
    depends_on:
      - main_db
    volumes:
      - ./flask_api:/usr/app
      - ./postgres/.pg/source/:/usr/app/src/static/
    env_file:
      - ./flask_api/.env
    expose:
      - 5000
    restart: always
    command: gunicorn --workers 4 --chdir usr/app --preload --bind 0.0.0.0:5000 --access-logfile - --error-logfile - main:app
    networks:
      - keykeeper-net
  nginx:
    container_name: nginx
    build: ./nginx
    #image: nginx:1.20-alpine
    ports:
      - 8181:80
      - 4343:443
    volumes:
      - ./static:/home/web/default/static
    depends_on: 
      - app
    restart: always
    networks:
      - keykeeper-net
  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4:8.5
    depends_on:
      - main_db
    ports:
      - "5550:5555/tcp"
    environment:
      - PGADMIN_DEFAULT_EMAIL=root@emoskvina.ru
      - PGADMIN_DEFAULT_PASSWORD=xe8ksbRVSex9 #При смене пересоздать контейнер
      - PGADMIN_LISTEN_ADDRESS=0.0.0.0
      - PGADMIN_LISTEN_PORT=5555
    networks:
      - keykeeper-net
  portainer: # H5p3RU_rbNRN
    container_name: portainer
    image: portainer/portainer-ce:latest
    ports:
      - "9000:9000/tcp"
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - portainer_data:/data
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - TZ=Europe/Moscow
      - VIRTUAL_PORT=9000
    restart: unless-stopped
    networks:
      - keykeeper-net

networks:
  keykeeper-net:
    driver: bridge

volumes:
  src:
  portainer_data: